src/excmds.ts

```diff
@@ -2664,7 +2666,191 @@ export async function viewcontainers() {
+            return tgroupActivate(name).then(() => undefined)
+        } else {
+            return tgroupcreate(name).then(() => undefined)
+        }
+    } else {
+        return tgroupcreate(name).then(() => undefined)
+    }
+}
+
+/**
+ * Switch to the previously active tab group.
+ */
+//#background
+export async function tgrouplast() {
+    if ((await tgroups()).size < 2) {
+        throw new Error("No last tab group")
+    }
+
+    return tgroupActivateLast().then(() => undefined)
+}
+
+/**
+ * Rename the current tab group.
+ *
+ * @param name The new name of the tab group.
+ *
+ */
+//#background
+export async function tgrouprename(name: string) {
+    if ((await tgroups()).size == 0) {
+        throw new Error("No tab groups exist")
+    }
+
+    return tgroupClearOldInfo(await windowTgroup() as string, name).then(() => undefined)
+}
+
+/**
+ * Close the current tab group.
+ *
+ * First switch to the previously active tab group. Do nothing if there is only
+ * one tab group.
+ *
+ */
+//#background
+export async function tgroupclose() {
+    const groups = await tgroups()
+    if (groups.size == 0) {
+        throw new Error("No tab groups exist")
+    } else if (groups.size == 1) {
+        throw new Error("This is the only tab group")
+    } else if (groups.size > 1) {
+        const closeGroup = await windowTgroup() as string
+        await tgroupActivateLast()
+        await tgroupRun(closeGroup).then(tabs => browser.tabs.remove(tabs.map(tab => tab.id)))
+        return tgroupClearOldInfo(closeGroup).then(() => undefined)
+    }
+}
+
+/**
+ * Move the current tab to another tab group.
+ *
+ * @param name The name of the tab group to move the tab to.
+ *
+ * If this is the last tab in the tab group, also switch to tab group, keeping
+ * the current tab active.
+ *
+ */
+//#background
+export async function tgroupmove(name: string) {
+    const groups = await tgroups()
+    const currentGroup = await windowTgroup() as string
+
+    if (groups.size == 0) {
+        throw new Error("No tab groups exist")
+    }
+    if (!groups.has(name)) {
+        throw new Error(`Tab group "${name}" does not exist`)
+    }
+    if (name == currentGroup) {
+        throw new Error(`Tab is already on group "${name}"`)
+    }
+
+    const tabCount = await tgroupRun(currentGroup).then(tabs => tabs.length)
+
+    await setTabTgroup(name)
+    const currentTabId = await activeTabId()
+
+    // switch to other group if this is the last tab in the current group
+    if (tabCount == 1) {
+        return Promise.all([
+            tgroupClearOldInfo(currentGroup),
+            setWindowTgroup(name),
+            setTabTgroup(name, currentTabId),
+            tgroupRun(name).then(tabs => browserBg.tabs.show(tabs.map(tab => tab.id))),
+        ]).then(() => undefined)
+    } else {
+        const lastTabId = await tgroupLastTabId(currentGroup)
+        await tabSetActive(lastTabId)
+        return browser.tabs.hide(currentTabId).then(() => undefined)
```

Comment:
I think there's something I don't understand here - as far as I can tell, this makes the currently active tab active and then hides it, without moving any tab anywhere. Is there something I'm missing?

Reply:
It's activating the previous tab not the current tab. It's already set the tab group name to the group it's being moved to, so all it needs to do is hide the tab. `:tgroupmove` stays on the current tab group except when run for the last tab in the tab group.